packages/amplify-category-function/src/provider-utils/awscloudformation/utils/storeResources.ts

```diff
@@ -139,15 +139,15 @@ export function updateLayerCfnFile(context, parameters, layerDirPath) {
     );
   }
   context.amplify.updateamplifyMetaAfterResourceUpdate(categoryName, parameters.layerName, 'runtimes', parameters.runtimes);
-  context.amplify.updateamplifyMetaAfterResourceUpdate(categoryName, parameters.layerName, 'versionsMap', parameters.layerVersionsMap);
+  context.amplify.updateamplifyMetaAfterResourceUpdate(categoryName, parameters.layerName, 'versionsMap', parameters.layerVersionMap);
   context.amplify.updateamplifyMetaAfterResourceUpdate(categoryName, parameters.layerName, 'build', parameters.build);
 }
 
 export function createLayerParametersFile(context, parameters, layerDirPath) {
   fs.ensureDirSync(layerDirPath);
   const parametersFilePath = path.join(layerDirPath, layerParametersFileName);
-  const jsonString = JSON.stringify({ parameters }, null, 4);
-  fs.writeFileSync(parametersFilePath, jsonString, 'utf-8');
+  //const jsonString = JSON.stringify({ parameters }, null, 4);
+  context.amplify.writeObjectAsJson(parametersFilePath, { parameters }, true);
```

Comment:
remove commented out code. Also, I don't think we need to be wrapping all the parameters in a top level "parameters" field (ie, remove the braces around parameters)